app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -94,24 +94,31 @@ class RecentSyncedTabFeature(
     }
 
     private suspend fun dispatchSyncedTabs() {
-        val syncedTab = storage.getSyncedDeviceTabs()
+        val syncedTabs = storage.getSyncedDeviceTabs()
             .filterNot { it.device.isCurrentDevice || it.tabs.isEmpty() }
-            .maxByOrNull { it.device.lastAccessTime ?: 0 }
+            .maxByOrNull { it.tabs.maxOf { tab -> tab.lastUsed } }
```

Comment:
Hi! I added this because of the bug mentioned in the issue. I could not find a bug filled for this specific case, so I'll just leave a comment with a reference to the issue. Since the devices are already ordered by the `lastUsed` tab, could we replace this line with `firstOrNull()`?

Reply:
I may be reading this incorrectly, but it looks like this logic will find the device with the most recent tab, and return all the tabs (up to the max) from that device specifically. I believe what we want is a list ordered strictly by `lastUsed`, but that may come from different devices. For example, if you had tabs on `device 1` that were 5 minutes and 15 minutes old, and tabs on `device 2` that were 10 and 20 minutes old it would go:

- device 1 (5 minute tab)
- device 2 (10 minute tab)
- device 1 (15 minute tab)
- device 2 (20 minute tab)

I will also update the issue to add some additional clarity.

Reply:
@MatthewTighe do you have a bug filed where you are seeing this behaviour? When the app services landed the `sortedByDescending` changes, it was meant to solve this problem already. We should really be filing the appropriate bug for the right team and at least link it here to remove our hack when it's solved.

Reply:
I think it's our issue actually, but I have not filed one yet. I'll add it to my list. I think it is because of this line in the code you linked above:

`fxaDevice?.let { SyncedDeviceTabs(fxaDevice, tabs.sortedByDescending { it.lastUsed }) }`

So we get a list of each device's list of tabs, rather than a flat list of tabs sorted by relative order.